src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationWriterILGen.cs

```diff
@@ -2416,6 +2416,67 @@ internal void ILGenForCreateInstance(CodeGenerator ilg, Type type, Type cast, bo
                 }
             }
 
+            Label labelReturn = ilg.DefineLabel();
+            Label labelEndIf = ilg.DefineLabel();
+
+            // Type typeInfo = type.GetTypeInfo();
+            LocalBuilder typeInfo = ilg.DeclareLocal(typeof(TypeInfo), "typeInfo");
+            ilg.Ldc(type);
+            MethodInfo getTypeInfoMehod = typeof(IntrospectionExtensions).GetMethod(
+                  "GetTypeInfo",
+                  CodeGenerator.StaticBindingFlags,
+                  new[] { typeof(Type) }
+                  );
+            ilg.Call(getTypeInfoMehod);
+
+            // IEnumerator e = typeInfo.DeclaredConstructors.GetEnumerator();
+            LocalBuilder enumerator = ilg.DeclareLocal(typeof(IEnumerator), "e");
+            MethodInfo getDeclaredConstructors = typeof(TypeInfo).GetMethod("get_DeclaredConstructors");
+            MethodInfo getEnumerator =
+                typeof(IEnumerable<>).MakeGenericType(typeof(ConstructorInfo)).GetMethod("GetEnumerator");
+            ilg.Call(getDeclaredConstructors);
+            ilg.Call(getEnumerator);
+            ilg.Stloc(enumerator);
+
+            ilg.WhileBegin();
+            // ConstructorInfo constructorInfo = e.Current();
+            MethodInfo enumeratorCurrent = typeof(IEnumerator).GetMethod("get_Current");
+            ilg.Ldloc(enumerator);
+            ilg.Call(enumeratorCurrent);
+            LocalBuilder constructorInfo = ilg.DeclareLocal(typeof(ConstructorInfo), "constructorInfo");
+            ilg.Stloc(constructorInfo);
+
+            // if (!constructorInfo.IsStatic || constructorInfo.GetParameters.Length() == 0)
```

Comment:
Shouldn't this be an && rather than ||?  I think the logic is right in the code, just not in the comment.